core/Network.js

```diff
@@ -5,10 +5,11 @@ var IIPConnection = require('./IIPConnection')
   , Process = require('./Process')
   , ProcessConnection = require('./ProcessConnection')
   , parseFBP = require('parsefbp')
-  , trace = require('./trace');
+  , trace = require('./trace')
+  , _ = require('lodash');
 
 var Network = module.exports = function () {
-  this._processes = [];
```

Comment:
This is what was killing you (in part).

JS uses `{ }` for objects and associative arrays. It uses `[ ]` for arrays.

When you use `foo['bar'] = 42` on a basic array, it's not strictly an error, but that value doesn't get put _into_ the array, so you get whacky behaviour.